### ISSUE TITLE
False positive: Could provoke a PHP Fatal error ([] operator not supported for strings).

### ISSUE THREAD

**Issue by anbuc** [kalessil/phpinspectionsea#1546]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 4.0.3 |
| Language level | PHP 7.3 |

Current behaviour
--
![image](https://user-images.githubusercontent.com/7095060/79775615-ced7ab00-8334-11ea-84bd-aea034b6be64.png)

The various different constellations where it works or not. 

Expected behaviour
--
No error reported

Environment details
---
```
PhpStorm 2020.1
Build #PS-201.6668.153, built on April 13, 2020
Runtime version: 11.0.6+8-b765.25 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0
GC: ParNew, ConcurrentMarkSweep
Memory: 4012M
Cores: 12
Registry: run.processes.with.pty=TRUE, debugger.valueTooltipAutoShowOnSelection=true, ide.tooltip.initialDelay=600
Non-Bundled Plugins: PlantUML integration, String Manipulation, com.tabnine.TabNine, lv.midiana.misc.phpstorm-plugins.deep-keys, com.kalessil.phpStorm.phpInspectionsEA, de.espend.idea.php.annotation, me.artspb.idea.eval.plugin, ru.adelf.idea.dotenv, org.sylfra.idea.plugins.linessorter
```

**Comment by kalessil** [comment]
Thank you for reporting @anbuc, this case is a known problem (IDE types solving limitations), so I have to close the issue. I recommend disabling the inspection, as it makes more distraction that any use in this context.
